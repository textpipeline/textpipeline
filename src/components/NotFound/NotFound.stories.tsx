import { storiesOf } from '@storybook/react';
import React from 'react';

import NotFound from './NotFound';

storiesOf('NotFound', module).add('default', () => <NotFound />);
