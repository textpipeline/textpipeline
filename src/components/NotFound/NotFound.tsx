import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import NotInterested from '@material-ui/icons/NotInterested';
import InternalLink from 'components/InternalLink';
import React from 'react';

const NotFound: React.FC = () => (
  <Box display="flex" flexDirection="column" justifyContent="space-evenly" alignItems="center">
    <Box my={5} display="flex" alignItems="center">
      <Box mx={1}>
        <NotInterested />
      </Box>
      <Typography variant="h4" component="h2" noWrap align="center">
        Not Found
      </Typography>
    </Box>
    <Typography component="p">That page doesn't seem to exist. It may have been moved or removed.</Typography>
    <InternalLink href="/">Click here to return home</InternalLink>
  </Box>
);

export default NotFound;
