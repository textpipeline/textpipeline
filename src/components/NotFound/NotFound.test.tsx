import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import NotFound from './NotFound';

describe('NotFound', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<NotFound />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
