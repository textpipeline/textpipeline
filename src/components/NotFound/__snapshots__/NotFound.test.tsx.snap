// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders correctly 1`] = `
<Styled(MuiBox)
  alignItems="center"
  display="flex"
  flexDirection="column"
  justifyContent="space-evenly"
>
  <Styled(MuiBox)
    alignItems="center"
    display="flex"
    my={5}
  >
    <Styled(MuiBox)
      mx={1}
    >
      <NotInterestedIcon />
    </Styled(MuiBox)>
    <WithStyles(ForwardRef(Typography))
      align="center"
      component="h2"
      noWrap={true}
      variant="h4"
    >
      Not Found
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
  <WithStyles(ForwardRef(Typography))
    component="p"
  >
    That page doesn't seem to exist. It may have been moved or removed.
  </WithStyles(ForwardRef(Typography))>
  <ForwardRef
    href="/"
  >
    Click here to return home
  </ForwardRef>
</Styled(MuiBox)>
`;
